bolt/src/Passes/Instrumentation.cpp

```diff
@@ -616,13 +616,24 @@ void Instrumentation::createAuxiliaryFunctions(BinaryContext &BC) {
     return Func;
   };
 
-  Summary->InitialIndCallHandlerFunction =
-      createSimpleFunction("__bolt_instr_default_ind_call_handler",
-                           BC.MIB->createInstrumentedNoopIndCallHandler());
+  BinaryFunction *IndCallHandler =
```

Comment:
It took me a while to understand what was happening here because we are not creating two functions but the real intent is to create two basic blocks of the same function that is coordinating entry to the code that performs instrumentation of indirect calls.

Can we rename this variable to IndCallHandlerExitBB, and add a comment explaning? Also, the methods "createInstrumentedIndCallHandler" and "createInstrumentedIndCallTrampoline" could be renamed to "createInstrumentedIndCalllHandlerExitBB" and "createInstrumentedIndCallHandlerEntryBB", respectively, so it's extra clear what we are doing here.

Regarding the comment explaining, I was thinking in something like:

// Here IndCallHandlerExitBB has the instructions to finish handling traffic
// to an indirect call. We pass it to createInstrumentedIndCallHandlerEntryBB(), which
// will decide between jumping to code in the runtime library (to account traffic)
// if the pointer is set, and then finish by jumping to the exit bb, which will return
// normal program execution.